builder/source/git.go

```diff
@@ -77,43 +77,33 @@ func (g *GitSource) submodules() error {
 	return cmd.Run()
 }
 
-// clone shallow clones an upstream git repository to the local disk.
-func clone(uri, path, ref string) error {
-	var cmd *exec.Cmd
-
-	// Check if the reference is a commit
-	if len(ref) == 40 {
-		// Init a new repository at the checkout path
-		initCmd := exec.Command("git", "init", path)
-
-		if err := initCmd.Run(); err != nil {
-			return err
-		}
+// For some reason git blobless clones create pack files with 600 permissions. These break future operations
+// as those files cannot be read by non-root users. Fix those permissions so things work as they should
+func (g *GitSource) fixPermissions() error {
+	cmd := exec.Command("bash", "-c", "chmod +r .git/objects/pack/*.promisor")
 
-		// Set the default remote to the upstream URI
-		addRemoteCmd := exec.Command("git", "remote", "add", "origin", uri)
-		addRemoteCmd.Dir = path
+	cmd.Dir = g.ClonePath
 
-		if err := addRemoteCmd.Run(); err != nil {
-			return err
-		}
+	return cmd.Run()
+}
 
-		// Shallow fetch the reference we want
-		fetchCmd := exec.Command("git", "fetch", "--depth", "1", "origin", ref)
-		fetchCmd.Dir = path
+// clone shallow clones an upstream git repository to the local disk.
+func clone(uri, path, ref string) error {
+	var cmd *exec.Cmd
 
-		if err := fetchCmd.Run(); err != nil {
-			return err
-		}
+	// Create a blobless clone without checking out a ref
+	initCmd := exec.Command("git", "clone", "--filter=blob:none", "--no-checkout", uri, path)
+	initCmd.Stdout = os.Stdout
+	initCmd.Stderr = os.Stderr
```

Comment:
Output of called commands should go to stdout:

```suggestion
	initCmd.Stderr = os.Stdout
```

(the `os.Stderr` below is a bug: #79)